src/go/k8s/api/redpanda/v1alpha2/common.go

```diff
@@ -0,0 +1,181 @@
+	Cert *SecretKeyRef `json:"certSecretRef,omitempty"`
+	// Key is the reference for client private certificate to establish mTLS connection to Redpanda
+	Key *SecretKeyRef `json:"keySecretRef,omitempty"`
+	// InsecureSkipTLSVerify can skip verifying Redpanda self-signed certificate when establish TLS connection to Redpanda
+	// +optional
+	InsecureSkipTLSVerify bool `json:"insecureSkipTlsVerify"`
+}
+
+// SecretKeyRef contains enough information to inspect or modify the referred Secret data
+// See https://pkg.go.dev/k8s.io/api/core/v1#ObjectReference.
+type SecretKeyRef struct {
+	// Name of the referent.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+	Name string `json:"name"`
+
+	// +optional
+	// Key in Secret data to get value from
+	Key string `json:"key,omitempty"`
+}
+
+// GetValue retrieves the value from `corev1.Secret{}`.
+func (s *SecretKeyRef) GetValue(
+	ctx context.Context, cl client.Client, namespace, defaultKey string,
+) ([]byte, error) {
+	secret := &corev1.Secret{}
+	if err := cl.Get(ctx, client.ObjectKey{Namespace: namespace, Name: s.Name}, secret); err != nil {
+		return nil, fmt.Errorf("getting Secret %s/%s: %w", namespace, s.Name, err)
+	}
+
+	b, err := s.getValue(secret, namespace, defaultKey)
+	if err != nil {
+		return nil, err
+	}
+
+	return b, nil
+}
+
+func (s *SecretKeyRef) getValue(
+	secret *corev1.Secret, namespace, defaultKey string,
+) ([]byte, error) {
+	key := s.Key
+	if key == "" {
+		key = defaultKey
+	}
+
+	value, ok := secret.Data[key]
+	if !ok {
+		return nil, fmt.Errorf("getting value from Secret %s/%s: key %s not found", namespace, s.Name, key) //nolint:goerr113 // no need to declare new error type
+	}
+	return value, nil
+}
+
+// AdminAPISpec defines client configuration for connecting to Redpanda's admin API.
+type AdminAPISpec struct {
+	// Specifies a list of broker addresses in the format <host>:<port>
+	URLs []string `json:"urls"`
+	// Defines TLS configuration settings for Redpanda clusters that have TLS enabled.
+	// +optional
+	TLS *CommonTLS `json:"tls,omitempty"`
+	// Defines authentication configuration settings for Redpanda clusters that have authentication enabled.
+	// +optional
+	SASL *AdminSASL `json:"sasl,omitempty"`
+}
+
+// AdminSASL configures credentials to connect to Redpanda cluster that has authentication enabled.
+type AdminSASL struct {
+	// Specifies the username.
+	// +optional
+	Username string `json:"username,omitempty"`
+	// Specifies the password.
+	// +optional
+	Password SecretKeyRef `json:"passwordSecretRef,omitempty"`
+	// Specifies the SASL/SCRAM authentication mechanism.
+	Mechanism SASLMechanism `json:"mechanism"`
+	// +optional
+	AuthToken SecretKeyRef `json:"token,omitempty"`
+}
+
+// ClusterRef represents a reference to a cluster that is being targeted.
+type ClusterRef struct {
+	// Name specifies the name of the cluster being referenced.
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
```

Comment:
Just FYI, I made this not just a simple string so that we can potentially extend the way we do "Refs" in the future and make them more flexible/follow pretty well-worn patterns in k8s land. We could potentially do something like add a `Namespace` for doing cross-namespace refs or add `Group`/`Kind` information to potentially support other CRD types in the future if we change the way we configure clusters with some alternative CRD structure.

Reply:
We could also re-use or embed `corev1.LocalObjectReference` if we wanted to leverage builtin types/existing documentation. No strong feelings, just sharing if you were unfamiliar with the the type.

Reply:
I'm thinking that embedding wouldn't get us much as it's literally just a single field and might feel a bit clunky to extend with additional fields in the future. What I'm envisioning we may want to support some time in the future is like halfway between a `LocalObjectReference` and a full-blown [`ObjectReference`](https://pkg.go.dev/k8s.io/api/core/v1#ObjectReference), something roughly of the shape:

```
Group     *string
Kind      *string
Namespace *string
Name      string
```